docs/architecture.md

```diff
@@ -112,76 +112,206 @@ The redux store is composed of two `slices` : `documents` and `queries`:
+Note that you can easily view each query using [Devtools](/docs/devtools.md) to identify duplicates.
+
+
+#### Complete query flow
+
+```mermaid
+---
+title: Query with cozy-client
+---
+sequenceDiagram
+    participant App as App
+    participant CC as cozy-client
+    participant Store as Store
+    participant Backend as Backend
+
+    App->>CC: Query documents
+    CC->>Store: Check query
+    alt Query does not exist
+    CC->>Store: Init query
+    Store->>Store: Add query to collection
+    else Query exists
+    CC->>CC: Check fetch policy
+    CC->>CC: Check loading status
+    end
+    CC->>Store:  Load query
+    Store->>Store: Update query status
+    CC->>Backend:  Execute query
+    Backend->>CC:  Data
+    CC->>Store: Data
+    Store->>Store: Update query status
+    Store->>Store: Update documents 
+    CC->>App: Results
+    Store->>Store: Auto-update queries
+```
+
+#### Persisted store and background fetching stategy 
+
+An app can decide to persist its store in localstorage. See [here](https://github.com/cozy/cozy-home/blob/c47858515d4f95c24fa88a6b96bb0b500b947424/src/store/configureStore.js#L46) for an example.
+When doing so, the app loads the store when it starts and benefit from the instant available cached data.
+
+But, the data could have significantly evolved between the last store save and the start, making the documents potentially stale. 
+Hopefully, since the fetch policy is probably expired, queries will be run again, retrieving fresh data. Unfortunately, this might cause an immediate spinning display on the app when it starts, even though there is available data thanks to the persisted store. 
+
+To solve this, a `backgroundFetching` option can be set, at the query level, or at the store level (enabled for all queries).
+
+When the query is loaded, it simply keeps the query status in `loaded` state (rather than setting a `loading` state) and set a new `isFetching` attribute, stored in the query. 
+Thanks to this, the app can adapt its display to avoid the "spinning effect", and might inform the user that data is being updated in the background, without refreshing the whole app.
+
+See [this PR](https://github.com/cozy/cozy-client/pull/1211) for more insights.
+
+
+Here is a simplified execution sequence with or without background fetching:
+```mermaid
+---
+title: Background fetching
+---
+sequenceDiagram
+    participant App as App
+    participant CC as cozy-client
+    participant Store as Store
+    participant Backend as Backend
+
+    App->>CC: Query documents
+
+    CC->>Store:  Load query
+    alt Without background fetching
+      Store->>Store: update query status
+      Store->>App: Loading query status
+      App->>App: Spinner
+    else With background fetching
+      Store->>Store: no change on query status
+      Store->>App: isFetching status
+      App->>App: Do nothing
```

Comment:
With `Do nothing` my understanding is that the component won't even display anything. Maybe something like `Display persisted data` would be more explicit?

Reply:
The "Do nothing" was to make an opposition to the "spinner" sceario, in reaction to the changing status. Here, no status change, so no need to do anything on the app side.
But ok to stress out the data displaying, I'll make the change